File: arrays/go/array4.go
package main

import "fmt"

func main(){
	arr1:=[]int{1, 2, 3, 4, 5}
	arr2:=[]int{2, 4, 6, 8} // o/p: 2, 4
    intersectionOfArrays(arr1, arr2)
}

func intersectionOfArrays(arr1 []int, arr2 []int){
	i, j, n1, n2:=0, 0, len(arr1), len(arr2)
	intersection:=make([]int, 0)
	for i<n1 && j<n2{
		if arr1[i]==arr2[j]{
			intersection=append(intersection, arr1[i])
			i++ 
			j++
		}else if arr1[i]<arr2[j] { i++ 
		}else { j++ }
	}
	fmt.Println(intersection)
	}

File: arrays/java/Array4.java
import java.util.*;


class Array4{
    public static void main(String[] args){
        int[] arr1={1, 2, 3, 4, 5};
	    int[] arr2={2, 4, 6, 8}; // o/p: 2, 4
        intersectionOfArrays(arr1, arr2);
    }
    static void intersectionOfArrays(int[] arr1, int[] arr2){
        int n1 = arr1.length;
        int n2 = arr2.length;
        List<Integer> intersection = new ArrayList<>();
        int i=0;
        int j=0;
        while(i<n1 && j<n2){
            if(arr1[i]==arr2[j]) {
                intersection.add(arr1[i]);
                i++;
                j++;
            }else if(arr1[i]<arr2[j]) i++;
            else j++;
        }
        System.out.println(intersection); 
    }
}